prost-derive/src/field/scalar.rs

```diff
@@ -555,13 +550,13 @@ impl Ty {
             Ty::Bool => quote!(bool),
             Ty::String => quote!(&str),
             Ty::Bytes(..) => quote!(&[u8]),
-            Ty::Enumeration(..) => quote!(i32),
+            Ty::Enumeration(..) => unreachable!("an enum should never be queried for its ref type"),
         }
     }
 
     pub fn module(&self) -> Ident {
         match *self {
-            Ty::Enumeration(..) => Ident::new("int32", Span::call_site()),
+            Ty::Enumeration(..) => Ident::new("enumeration", Span::call_site()),
```

Comment:
Does it make sense to use `self.as_str()` for enumerations as well?

Reply:
No, that would give "enum", but we can't use that as a module name. OTOH, maybe the current behavior of `Ty::as_str` should be changed: "enum" is not a real type name anyway.